src/backend/utils.ts

```diff
@@ -969,14 +1005,47 @@ export async function downloadDefaultWine() {
   return null
 }
 
+export async function setGPTKDefaultOnMacOS() {
+  const isGPTKCompatible = await isMacSonomaOrHigher()
+  if (!isGPTKCompatible) {
+    return
+  }
+
+  const wineList = await GlobalConfig.get().getAlternativeWine()
+  const gptk = wineList.find((wine) => wine.type === 'toolkit')
+  if (!gptk) {
+    await downloadDefaultWine()
+    return
+  }
+  if (gptk && existsSync(gptk.bin)) {
+    logInfo(`Changing wine version to ${gptk.name}`)
+    GlobalConfig.get().setSetting('wineVersion', gptk)
+    // update prefix to use the new one as well
+    const installPath = GlobalConfig.get().getSettings().defaultInstallPath
+    const newPrefix = join(installPath, 'Prefixes', 'GPTK')
+    GlobalConfig.get().setSetting('winePrefix', newPrefix)
+
+    return true
+  }
+  return false
+}
+
 export async function checkWineBeforeLaunch(
   appName: string,
   gameSettings: GameSettings,
   logFileLocation: string
 ): Promise<boolean> {
   const wineIsValid = await validWine(gameSettings.wineVersion)
 
-  if (wineIsValid) {
+  const isToolkit = gameSettings.wineVersion.type === 'toolkit'
+  const isGPTKCompatible = await isMacSonomaOrHigher()
+
+  const isValidOnLinux = isLinux && wineIsValid
+  const isValidOnMac =
```

Comment:
Things got a bit convoluted here but I am not sure if there is a better way to write this code.

Reply:
could maybe break these up into two variables `isMac && isToolkit && isGPTKCompatible && wineIsValid` and `isMac && !isToolkit && wineIsValid`. I'm not sure what you'd name them. could add a comment too if it still seems unclear